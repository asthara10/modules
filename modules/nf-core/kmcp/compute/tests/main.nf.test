nextflow_process {

    name "Test Process KMCP_COMPUTE"
    script "../main.nf"
    process "KMCP_COMPUTE"

    tag "modules"
    tag "modules_nfcore"
    tag "kmcp"
    tag "kmcp/compute"

    test("test_kmcp_compute") {


        when {
            process {
                """
                
                input[0] = [
        [ id:'test', single_end:false ], // meta map
        file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_kmcp_compute_directory") {
        when {
            process {
                """
                        input[0] = UNTAR ( [
            [ id:'test' ],
            file("https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/kmcp/kmcp_profile.tar.gz", checkIfExists: true)
        ]).untar


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }
    }
    
}
