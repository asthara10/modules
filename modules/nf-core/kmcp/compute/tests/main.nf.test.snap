{
    "versions": {
        "content": [
            [
                "versions.yml:md5,e564e902249d469b20b45233db5cccaf"
            ]
        ],
        "timestamp": "2023-11-24T13:31:09.011734"
    }
}